pkg/analysis_server/lib/src/services/correction/fix_internal.dart

```diff
@@ -1875,6 +1884,25 @@ class FixProcessor {
     }
   }
 
+  void _addFix_removeTypeName() {
+    final FunctionTypedFormalParameter functionTyped =
+        node is FunctionTypedFormalParameter
+            ? node
+            : node.getAncestor((node) => node is FunctionTypedFormalParameter);
+    if (functionTyped != null) {
+      _addReplaceEdit(rf.rangeNode(functionTyped),
+          utils.getNodeText(functionTyped.identifier));
+      _addFix(DartFixKind.REMOVE_TYPE, []);
+      return;
+    }
```

Comment:
Looking at the code for `AvoidAnnotatingWithDynamic` it looks like it will visit every `SimpleFormalParameter`, even ones inside a function typed formal parameter.

(As a side-note, adding a passing test that proves that we get the right behavior would probably shorten this conversation :-)